src/api/PDFDocumentOptions.ts

```diff
@@ -35,4 +35,5 @@ export interface CreateOptions {
 export interface EmbedFontOptions {
   subset?: boolean;
   customName?: string;
+  features?: any[];
```

Comment:
I prefer to avoid using the `any` type in the pdf-lib codebase. It prevents users from being able to easily understand what types of features can actually be controlled. Can we use a specific type here that defines what features are supported? E.g. `features: { foo: boolean, bar: string }`.

Reply:
First, let me get it out my chest: I'm new to TS. 🙈

That being said, I've tried to set the type like so: `string[] | {[key: string]: boolean}`, which theoretically is the type that param can be, but  I got errors as fontKit set that type to `any[]` (hence why I used it in the first place). So we could either leave like fontKit set it or choose one type only like `string[]` which I didn't get any errors. What would you suggest?

PS: I can see why they did like that, there's a long list of possible features (https://docs.microsoft.com/en-gb/typography/opentype/spec/featurelist)

Reply:
No worries! Appreciate you contributing despite being new :)

`fontkit` doesn't actually have any TS types. `pdf-lib` uses a fork called `@pdf-lib/fontkit`, which has its own types defined [here](https://github.com/Hopding/fontkit/blob/master/fontkit.d.ts). I created them based on the `fontkit` README, which does not document the ability to pass objects in addition to arrays. We probably need to update the [`@pdf-lib/fontkit/fontkit.d.ts`](https://github.com/Hopding/fontkit/blob/master/fontkit.d.ts) file to accept both objects and arrays (and really they shouldn't be `any` types either, if we know it's just strings and booleans that are allowed). Once that's done, we should be able to use the `string[] | {[key: string]: boolean}` type without any issues.

I would prefer to have an interface that essentially represents the table in https://docs.microsoft.com/en-gb/typography/opentype/spec/featurelist. It would be nice for autocomplete, typechecking, and documentation purposes. However, if you prefer to just have `string[] | {[key: string]: boolean}`, then I'm fine with that too.